source/AndroidResolver/src/GradleResolver.cs

```diff
@@ -130,6 +130,54 @@ public class GradleResolver
             return sourcesByPackageSpec;
         }
 
+        /// <summary>
+        /// From a list of dependencies generate a list of Maven / Gradle / Ivy package spec
+        /// strings along with their corresponding dependency versions.
+        /// </summary>
+        /// <param name="dependencies">Dependency instances to query for package specs.</param>
+        /// <returns>Dictionary where the key is a package spec string and value is a
+        /// 3 item list of sources string, dependency version less string,
+        /// and a dependency version string. </returns>
+        internal static Dictionary<string, List<string>> DependenciesToPackageSpecsWithVersions(
+                IEnumerable<Dependency> dependencies) {
+            // To keep it simple to iterate and process, create 3 separate
+            // dictionaries to record required metadata about a package spec
+            // (version and versionless key).
+            var sourcesByPackageSpec = new Dictionary<string, string>();
+            var versionsByPackageSpec = new Dictionary<string, string>();
+            var versionlessKeysByPackageSpec = new Dictionary<string, string>();
+            foreach (var dependency in dependencies) {
+                // Convert the legacy "LATEST" version spec to a Gradle version spec.
+                var packageSpec = dependency.Version.ToUpper() == "LATEST" ?
+                    dependency.VersionlessKey + ":+" : dependency.Key;
+                var source = CommandLine.SplitLines(dependency.CreatedBy)[0];
+                string sources;
+                if (sourcesByPackageSpec.TryGetValue(packageSpec, out sources)) {
+                    sources = sources + ", " + source;
+                } else {
+                    sources = source;
+                }
+                sourcesByPackageSpec[packageSpec] = sources;
+                versionsByPackageSpec[packageSpec] = dependency.Version;
+                versionlessKeysByPackageSpec[packageSpec] = dependency.VersionlessKey;
+            }
+            // Create a new dictionary which combines the data from sources and
+            // versions dictionaries. Key is packageSpec string and value is a
+            // three-item list of strings of sources, version less string
+            // and dependency version.
+            // NOTE: tuples are a more elegant way of doing this but they work
+            // only in .NET frameworks >= 4.0.
+            var mergedByPackageSpec = new Dictionary<string, List<string>>();
+            foreach(var item in sourcesByPackageSpec) {
```

Comment:
You do not need a separate loop.  You can construct mergedByPackageSpec in the previous loop.

Reply:
Ended up following your other suggestion and not needing this code.